pkg/tests/testinfra/testinfra.go

```diff
@@ -23,6 +24,9 @@ import (
 // StartGrafana starts a Grafana server.
 // The server address is returned.
 func StartGrafana(t *testing.T, grafDir, cfgPath string, sqlStore *sqlstore.SQLStore) string {
+	// Grafana uses some global metric registration. Clear the
+	// registry between tests.
+	prometheus.DefaultRegisterer = prometheus.NewRegistry()
 	t.Helper()
```

Comment:
Nit, but better to keep `t.Helper` at the very beginning of the function, because it matters when the stacktrace is calculated.
```suggestion
	t.Helper()
	// Grafana uses some global metric registration. Clear the
	// registry between tests.
	prometheus.DefaultRegisterer = prometheus.NewRegistry()
```

Reply:
Gotcha,  thanks!